stubs/urllib3/urllib3/util/retry.pyi

```diff
@@ -12,16 +12,25 @@ ResponseError = exceptions.ResponseError
 log: Any
 
 class Retry:
-    DEFAULT_METHOD_WHITELIST: Any
-    BACKOFF_MAX: Any
-    total: Any
-    connect: Any
-    read: Any
-    redirect: Any
-    status_forcelist: Any
-    method_whitelist: Any
-    backoff_factor: Any
-    raise_on_redirect: Any
+    DEFAULT_ALLOWED_METHODS: ClassVar[frozenset[str]]
+    RETRY_AFTER_STATUS_CODES: ClassVar[frozenset[int]]
+    DEFAULT_REMOVE_HEADERS_ON_REDIRECT: ClassVar[frozenset[str]]
+    DEFAULT_BACKOFF_MAX: ClassVar[int]
+
+    total: bool | int | None
+    connect: int | None
+    read: int | None
+    redirect: bool | int | None
```

Comment:
It appears that redirect can never be `False`, only `int`, `None` or `True`. You'll need to import `Literal` from `typing_extensions`.

```suggestion
    redirect: Literal[True] | int | None
```